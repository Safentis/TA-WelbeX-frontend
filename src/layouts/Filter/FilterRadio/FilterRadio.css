.filter-radio {
 
}

.filter-radio__item {

}

.filter-radio__label {
  color: var(--main-dark-blue-color);
  cursor: pointer;
  font-size: 2rem;
  margin: 0 1rem;
  text-transform: uppercase;
}

.filter-radio__label_active {
  color: var(--main-purple-color);
}

.filter-radio__label:hover {
  opacity: 0.5;
  color: var(--main-purple-color);
}

.filter-radio__input {
  opacity: 0;
  position: absolute;
  z-index: -10;
}
