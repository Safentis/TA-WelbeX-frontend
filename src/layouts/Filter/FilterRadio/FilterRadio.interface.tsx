import React from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-common-types';

export interface Props {
  className?: string;
  icons?: IconDefinition[];
  name: string;
  fields: {
    [key: string]: boolean;
  };
  handleChange: (event: React.ChangeEvent) => void;
}
