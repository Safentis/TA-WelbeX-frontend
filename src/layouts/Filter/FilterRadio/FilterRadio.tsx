import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React, { FC } from 'react';

import './FilterRadio.css';
import { Props } from './FilterRadio.interface';

/**
 * FilterRadio component, that contains radio - inputs, which set up filter
 * @param {string} className - no required,
 * @param {string} name - required, name of group inputs
 * @param {[key: string]: boolean} fields - required,
 * object with boolean field
 * @param {(event: React.ChangeEvent) => void} handleChange - required,
 * function, that handle radio inputs
 * @returns {React.ReactElement} - returns react element
 */

//prettier-ignore
const FilterRadio: FC<Props> = ({ className = "", name, fields, handleChange, icons = [] }): React.ReactElement => {
  const entries = Object.entries(fields); //* Created entries from a common object
  const content = entries.map(([key, value], index: number) => (
    <li className="filter-radio__item" key={index}>
      <label 
        className={`filter-radio__label ${value ? "filter-radio__label_active" : ""}`} 
        htmlFor={key}
      >
        {
          icons.length === 0 
            ? key
            : <FontAwesomeIcon icon={icons[index]}/> 
        }
        <input
          className="filter-radio__input"
          name={name}
          type="radio"
          value={key}
          id={key}
          onChange={handleChange}
          checked={value}
        />
      </label>
    </li>
  ));
  return <ul className={`filter-radio ${className}`}>{content}</ul>;
};

export default FilterRadio;
