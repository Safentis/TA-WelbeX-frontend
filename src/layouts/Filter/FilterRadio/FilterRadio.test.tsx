import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import FilterRadio from './FilterRadio';

import { Props } from './FilterRadio.interface';

const setUp = (props: Props) => {
  return shallow(<FilterRadio {...props} />);
};

describe('<FilterRadio />', () => {
    let component: ShallowWrapper<
      any,
      Readonly<{}>,
      React.Component<{}, {}, any>
    >;
    let props = {
      className: 'custom-style',
      handleChange: () => {},
      fields: {
        name: true,
        number: false,
        distance: false,
      },
    }
  
    beforeEach(() => {
      component = setUp(props);
    });
  
    it('component does match to the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  
    describe('Markup: ', () => {
      it('has 1 ul element', () => {
        expect(component.find('ul')).toBeTruthy();
      });
  
      it('has 3 li elements', () => {
        expect(component.find('li').length === 3).toBeTruthy();
      });
  
      it('has 3 input element', () => {
        expect(component.find('input').length === 3).toBeTruthy();
      });
      
      it('has 3 label element', () => {
        expect(component.find('label').length === 3).toBeTruthy();
      });
    });
  });
  