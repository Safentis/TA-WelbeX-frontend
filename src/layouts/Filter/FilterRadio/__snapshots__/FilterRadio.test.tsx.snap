// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterRadio /> component does match to the snapshot 1`] = `
<ul
  className="filter-radio custom-style"
>
  <li
    className="filter-radio__item"
    key="0"
  >
    <label
      className="filter-radio__label filter-radio__label_active"
      htmlFor="name"
    >
      name
      <input
        checked={true}
        className="filter-radio__input"
        id="name"
        name="column"
        onChange={[Function]}
        type="radio"
        value="name"
      />
    </label>
  </li>
  <li
    className="filter-radio__item"
    key="1"
  >
    <label
      className="filter-radio__label "
      htmlFor="number"
    >
      number
      <input
        checked={false}
        className="filter-radio__input"
        id="number"
        name="column"
        onChange={[Function]}
        type="radio"
        value="number"
      />
    </label>
  </li>
  <li
    className="filter-radio__item"
    key="2"
  >
    <label
      className="filter-radio__label "
      htmlFor="distance"
    >
      distance
      <input
        checked={false}
        className="filter-radio__input"
        id="distance"
        name="column"
        onChange={[Function]}
        type="radio"
        value="distance"
      />
    </label>
  </li>
</ul>
`;
